src/sentry/static/sentry/app/views/settings/components/forms/controls/textarea.tsx

```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextareaAutosize from 'react-autosize-textarea';
+import styled from '@emotion/styled';
+import isPropValid from '@emotion/is-prop-valid';
+
+import {inputStyles} from 'app/styles/input';
+
+type InputProps = Omit<Parameters<typeof inputStyles>[0], 'theme'>;
+type Props = React.TextareaHTMLAttributes<HTMLTextAreaElement> &
+  InputProps & {
+    /**
+     * Enable autosizing of the textarea.
+     */
+    autosize?: boolean;
+    /**
+     * Number of rows to default to.
+     */
+    rows?: number;
+  };
+
+const TextAreaControl = React.forwardRef<HTMLTextAreaElement, Props>(
+  ({autosize, rows, ...p}, ref) =>
```

Comment:
if you add the `Props` here, you won't need to declare the `propTypes`
```suggestion
  ({autosize, rows, ...p}: Props, ref: React.Ref< HTMLTextAreaElement >) =>
```